File: Samples-Java/Samples/src/main/java/fragments/autocomplete/AutoCompleteCustomizationFragment.java
package fragments.autocomplete;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.Display;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.telerik.android.sdk.R;
import com.telerik.widget.autocomplete.AutoCompleteAdapter;
import com.telerik.widget.autocomplete.CompletionMode;
import com.telerik.widget.autocomplete.DisplayMode;
import com.telerik.widget.autocomplete.RadAutoCompleteTextView;
import com.telerik.widget.autocomplete.SuggestMode;
import com.telerik.widget.autocomplete.TokenModel;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import activities.ExampleFragment;

public class AutoCompleteCustomizationFragment extends JsonDataLoadFragment implements ExampleFragment {

    private JSONArray data;
    private RadAutoCompleteTextView autocomplete;
    private AutoCompleteAdapter adapter;

    @Override
    public String title() {
        return "Customization";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.autocomplete_customization, container, false);

        autocomplete = (RadAutoCompleteTextView) rootView.findViewById(R.id.autocomplete);
        autocomplete.setSuggestMode(SuggestMode.SUGGEST);
        autocomplete.setDisplayMode(DisplayMode.TOKENS);

        String jsonData = this.getJSONFile(R.raw.countries);
        try{
            JSONObject jObj = new JSONObject(jsonData);
            data = jObj.getJSONArray("data");
        }
        catch(JSONException ex){
            ex.printStackTrace();
        }

        adapter = new AutoCompleteAdapter(this.getContext(),this.getTokenModelObjects(data), R.layout.suggestion_item_layout);
        adapter.setCompletionMode(CompletionMode.CONTAINS);
        autocomplete.setAdapter(adapter);

        Display display = this.getActivity().getWindowManager().getDefaultDisplay();
        int height  =  display.getHeight();
        autocomplete.setSuggestionViewHeight(height/3);
        Drawable img = getResources().getDrawable(R.drawable.search);
        autocomplete.setAutocompleteIcon(img);

        return rootView;
    }

    private ArrayList<TokenModel> getTokenModelObjects(JSONArray json) {
        ArrayList<TokenModel> feedData = new ArrayList<TokenModel>();
        JSONObject current = new JSONObject();
        for(int i = 0; i < json.length(); i++) {
            String name = "";
            String flag = "";

            try{
                current = json.getJSONObject(i);
                name = (String)current.get("country");
                flag = (String)current.get("flag");
            }
            catch (JSONException ex){
                ex.printStackTrace();
            }

            int dr = getResources().getIdentifier(flag, "drawable","com.telerik.android.sdk");
            Drawable m = getResources().getDrawable(dr);
            TokenModel token = new TokenModel(name,m, null);
            feedData.add(token);
        }

        return feedData;
    }
}

File: Samples-Xamarin/Fragments/Autocomplete/AutoCompleteCustomizationFragment.cs

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics.Drawables;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.Autocomplete;
using Newtonsoft.Json;

namespace Samples
{
	public class AutoCompleteCustomizationFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		private RadAutoCompleteTextView autocomplete;
		private AutoCompleteAdapter adapter;

		public string Title()
		{
			return "Customization";
		}

		public override Android.Views.View OnCreateView(Android.Views.LayoutInflater inflater, Android.Views.ViewGroup container, Android.OS.Bundle savedInstanceState)
		{
			View rootView = inflater.Inflate(Resource.Layout.fragment_autocomplete_customization, container, false);
			this.autocomplete = (RadAutoCompleteTextView)rootView.FindViewById(Resource.Id.autocomplete);

			List<FeedObject> objects = this.loadDataFromJson();

			// >> autocomplete-suggest-mode-xamarin
			this.autocomplete.SuggestMode = SuggestMode.Suggest;
			// << autocomplete-suggest-mode-xamarin

			this.autocomplete.DisplayMode = DisplayMode.Tokens;

			this.adapter = new AutoCompleteAdapter(this.Context, this.GetTokenObjects(objects), Java.Lang.Integer.ValueOf(Resource.Layout.suggestion_item_layout));
			this.adapter.CompletionMode = CompletionMode.StartsWith;
			this.autocomplete.Adapter = this.adapter;

			Display display = this.Activity.WindowManager.DefaultDisplay;
			int height = display.Height;
			this.autocomplete.SuggestionViewHeight = height / 3;

			Drawable img = Resources.GetDrawable(Resource.Drawable.search);
			this.autocomplete.SetAutocompleteIcon(img);

			return rootView;
		}

		private List<TokenModel> GetTokenObjects(List<FeedObject> initialObjects)
		{
			List<TokenModel> feedData = new List<TokenModel>();
			for (int i = 0; i < initialObjects.Count; i++)
			{
				int dr = Resources.GetIdentifier(initialObjects[i].Flag, "drawable", "com.telerik.android.sdk.xamarin");
				Drawable drawable = Resources.GetDrawable(dr);
				TokenModel token = new TokenModel(initialObjects[i].Country, drawable);
				feedData.Add(token);
			}

			return feedData;
		}

		private List<FeedObject> loadDataFromJson()
		{
			String json = LoadJSONFromAsset("countries.json");
			List<FeedObject> feed = JsonConvert.DeserializeObject<List<FeedObject>>(json);
			return feed;
		}

		private String LoadJSONFromAsset(String fileName)
		{
			String json = null;
			try
			{
				using (StreamReader sr = new StreamReader(Activity.Assets.Open(fileName)))
				{
					json = sr.ReadToEnd();
				}
			}
			catch (IOException ex)
			{
				Log.Error("error", ex.StackTrace);
			}
			return json;
		}
	}
}
